Odm upload

jacobrousseau opened the pull request:
The first version of the ODM-upload. Included are:
- fix of the problem encountered by Bo and Rusland due to a problem with the JaxBContext code
- mapping of the subjectLabel to the subjectOID; unfortunately at this moment we need N calls the the OpenClinica WS isStudySubject
- small fixes

jacobrousseau committed:
Ongoing work on the autonomous upload: finished the automatic scheduling

jacobrousseau committed:
Addtion of the stub of the FileCopyService

jacobrousseau committed:
Added the first version of the FileCopyService which manages the file traffic of the
autonomous upload

jacobrousseau committed:
Ongoing work. Removed the workdirectory and addtion of a timestamp to the
file name when copying it to the 'completed' directory

jacobrousseau committed:
Merge remote-tracking branch 'upstream/master' into AutonumousTask

jacobrousseau committed:
Ongoing work on the validation of the CRF-version present in the study compared against the
version(s) in the data

jacobrousseau committed:
Check-in for validation of CRF-version in data compared to the existing CRF-version(s) in
OpenClinica. Changed the list of ItemDefinitions from a java.util.List to a java.util.Set and added a equals / hash to
eliminiate duplicate items.

jacobrousseau committed:
Fix of the broken build.

jacobrousseau committed:
Merge remote-tracking branch 'upstream/master' into odm-upload

Conflicts:
	build.gradle
	src/main/generated/org/openclinica/ws/beans/ObjectFactory.java
	src/main/generated/org/openclinica/ws/studysubject/v1/ObjectFactory.java
	src/main/generated/org/openclinica/ws/studysubject/v1/Ws.java
	src/main/generated/org/openclinica/ws/studysubject/v1/WsService.java
	src/main/java/nl/thehyve/ocdu/OcduApplication.java
	src/main/java/nl/thehyve/ocdu/services/OpenClinicaService.java
	src/main/java/nl/thehyve/ocdu/services/ValidationService.java
	src/main/java/nl/thehyve/ocdu/soap/SOAPRequestFactory.java
	src/main/java/nl/thehyve/ocdu/validators/ClinicalDataOcChecks.java
	src/main/java/nl/thehyve/ocdu/validators/crossChecks/CRFVersionMatchCrossCheck.java
	src/main/resources/application.properties

jacobrousseau committed:
Ongoing work on the conversion to ODM.

Commit:
ODM generation: generation finished. Still a couple of TODO's

Commit:
Merge remote-tracking branch 'upstream/master' into odm-upload

Conflicts:
	src/main/generated/org/openclinica/ws/beans/EventResponseType.java
	src/main/generated/org/openclinica/ws/beans/ObjectFactory.java
	src/main/generated/org/openclinica/ws/data/v1/ObjectFactory.java
	src/main/generated/org/openclinica/ws/data/v1/Ws.java
	src/main/generated/org/openclinica/ws/data/v1/WsService.java
	src/main/generated/org/openclinica/ws/event/v1/Ws.java
	src/main/generated/org/openclinica/ws/event/v1/WsService.java
	src/main/generated/org/openclinica/ws/study/v1/ObjectFactory.java
	src/main/generated/org/openclinica/ws/study/v1/Ws.java
	src/main/generated/org/openclinica/ws/study/v1/WsService.java
	src/main/generated/org/openclinica/ws/studyeventdefiniton/v1/Ws.java
	src/main/generated/org/openclinica/ws/studyeventdefiniton/v1/WsService.java
	src/main/generated/org/openclinica/ws/studysubject/v1/ObjectFactory.java
	src/main/generated/org/openclinica/ws/studysubject/v1/Ws.java
	src/main/generated/org/openclinica/ws/studysubject/v1/WsService.java

Commit:
Fix of problem with the listAllByStudy WS-call.

Commit:
Merge branch 'master' of https://github.com/thehyve/Open-Clinica-Data-Uploader into odm-upload

PiotrZakrzewski commented on src/main/resources/application.properties:
```diff
@@ -12,4 +12,10 @@ logging.level.org.springframework.web=DEBUG
+
+
+autonomous.cron.schedule=1 * * * * *
+autonomous.upload.source.directory=/home/jacob/tomcat/upload/source
```
This path is specific to your setup - do you think we could replace it with some relative path that will work also on the server and our workstations?

PiotrZakrzewski commented on src/test/java/nl/thehyve/ocdu/ODMGenerationTests.java:
```diff
@@ -0,0 +1,77 @@
+        List<ClinicalData> correctClinicalData = factory.createClinicalData(testODMGenerationCorrect);
+        clinicalDataOcChecks = new ClinicalDataOcChecks(metaData, correctClinicalData, testSubjectWithEventsTypeList);
+        ODMService odmService = new ODMService();
+        String result = odmService.generateODM(correctClinicalData, metaData);
```
It is probably pretty hard to check XML for correctness in the test. For now maybe we could start with a simple assertion checking if generated XML is well-formed (which would implicitly tests that it is not null and not empty as well). I found this nice SO question which could help us do it: http://stackoverflow.com/questions/6362926/xml-syntax-validation-in-java

Commit:
Ongoing work on ODM-generation:
- addtion of a call to isStudySubject WS
- addtion of the mapping of the subject-label to the subjectOID
- addtion of a toString to SOAPUtils

Commit:
Addition of XML escaping
